wicket-core/src/main/java/org/apache/wicket/protocol/http/mock/MockHttpSession.java

```diff
@@ -146,7 +145,8 @@ public ServletContext getServletContext()
 	@Override
 	public void invalidate()
 	{
-		Session session = (Session) attributes.get("wicket:" + BaseWicketTester.TestFilterConfig.class.getName() + ":session");
+		Session session = (Session) attributes.get("wicket:" + "BaseWicketTester.TestFilterConfig" + ":session"); // TODO
```

Comment:
Copied from previous PR #659:

>>> wicket-core/src/main/java/org/apache/wicket/protocol/http/mock/MockHttpSession.java

>> I think this class should be moved to wicket-tester

Ok, we could do that (I also thought about moving the whole  _package _org/apache/wicket/protocol/http/mock_ to wicket-tester but org.apache.wicket.core.util.string.ComponentRenderer in /src/main/java makes use of org.apache.wicket.protocol.http.mock.MockServletContext, so that won’t work).

Should we move only MockHttpSession to wicket-tester then? If not, how to rewrite this line in MockHttpSession.invalidate()?

Reply:
If we want to move the entire package I don't see many alternatives,,,maybe the best option first glance could be move it into wicket-util module (i.e org.apache.wicket.util.protocol.http.mock ). Waiting for better ideas :-)

Reply:
I don't think that moving the package from wicket-core to wicket-util would have advantages.

My initial assumption was that mocking stuff should be only test scoped, but it currently isn't (as ComponentRenderer shows). IMO we should leave it as it is (possibly requiring a better fix in MockHttpSession.java, see diff) OR move just MockHttpSession.java to wicket-tester?

Reply:
I think these classes should stay in -core.

Reply:
Ok, and what about _Session session = (Session) attributes.get("wicket:" + BaseWicketTester.TestFilterConfig.class.getName() + ":session");_? Where is this relevant?

Reply:
This seems to be the only open issue of this PR. Could someone please take a closer look at this?

(I just don't see any reference where this attribute "wicket:[class]:session" is set. Related could be AbstractPersistentPageStore.KEY_PREFIX = "wicket:", but I dont see any other ":session" in the code.)

Reply:
See org.apache.wicket.session.HttpSessionStore#getSessionAttributePrefix

Reply:
> See org.apache.wicket.session.HttpSessionStore#getSessionAttributePrefix

Ok, I see you changed this
originally:  ... _attributes.get("wicket:" + BaseWicketTester.TestFilterConfig.class.getName() + ":session");_
to: ... _attributes.get("wicket:org.apache.wicket.util.tester.BaseWicketTester.TestFilterConfig:session");_

I wonder if that's correct since TestFilterConfig is an inner class?

A simple alternative would be to save TestFilterConfig as new file and adapt the line to _attributes.get("wicket:org.apache.wicket.util.tester.TestFilterConfig:session");_

Reply:
Good catch!
![Screenshot from 2023-10-12 16-18-31](https://github.com/apache/wicket/assets/232002/c7574040-f91a-4f09-9a25-c6751b5bf160)

Reply:
I fixed the bug!
The best would be to use a real class reference instead of hardcoding a String like now.